core/primitives/src/utils.rs

```diff
@@ -73,19 +75,45 @@ impl KeyForAccount {
     }
 }
 
+#[derive(derive_more::AsRef, derive_more::Into)]
+#[as_ref(forward)]
+pub struct KeyForAccountId(Vec<u8>);
+
+impl KeyForAccountId {
+    #[inline]
+    pub fn estimate_len() -> usize {
+        KeyForColumnAccountId::estimate_len(col::ACCOUNT_ID)
+    }
+
+    pub fn new(account_id: &AccountId) -> Self {
+        let key = KeyForColumnAccountId::with_capacity(col::ACCOUNT_ID, account_id, 0);
+        Self(key.into())
+    }
+
+    pub fn from_hash(account_hash: CryptoHash) -> Self {
+        let mut key = Vec::with_capacity(Self::estimate_len());
+        key.extend(col::ACCOUNT_ID);
+        key.extend(account_hash.as_ref());
+        debug_assert_eq!(key.len(), Self::estimate_len());
+        Self(key)
+    }
+}
+
 #[derive(derive_more::AsRef, derive_more::Into)]
 #[as_ref(forward)]
 pub struct KeyForAccessKey(Vec<u8>);
 
 impl KeyForAccessKey {
-    fn estimate_prefix_len(account_id: &AccountId) -> usize {
-        KeyForColumnAccountId::estimate_len(col::ACCESS_KEY, account_id) + col::ACCESS_KEY.len()
+    #[inline]
+    fn estimate_prefix_len() -> usize {
+        KeyForColumnAccountId::estimate_len(col::ACCESS_KEY) + col::ACCESS_KEY.len()
```

Comment:
If you want to get rid of all the separators, there is one left here (it seems that `col::ACCESS_KEY` used to be used before and after the account_id).

Reply:
Oh. I see. Good point.